src/test/java/edu/kis/powp/jobs2d/events/SelectTestFigureOptionListener.java

```diff
@@ -3,23 +3,41 @@
 import java.awt.event.ActionEvent;
 import java.awt.event.ActionListener;
 
+import edu.kis.powp.jobs2d.command.ComplexCommandFactory;
 import edu.kis.powp.jobs2d.drivers.DriverManager;
 import edu.kis.powp.jobs2d.magicpresets.FiguresJoe;
 
 public class SelectTestFigureOptionListener implements ActionListener {
 
-	private DriverManager driverManager;
+	private final DriverManager driverManager;
 
 	public SelectTestFigureOptionListener(DriverManager driverManager) {
 		this.driverManager = driverManager;
 	}
 
 	@Override
 	public void actionPerformed(ActionEvent e) {
-		if (e.getActionCommand().equals("Figure Joe 1"))
-			FiguresJoe.figureScript1(driverManager.getCurrentDriver());
-		else if (e.getActionCommand().equals("Figure Joe 2"))
-			FiguresJoe.figureScript2(driverManager.getCurrentDriver());
+		ComplexCommandFactory complexCommandFactory;
+		switch (e.getActionCommand()) {
+			case "Figure Joe 1":
+				FiguresJoe.figureScript1(driverManager.getCurrentDriver());
+				break;
+			case "Figure Joe 2":
+				FiguresJoe.figureScript2(driverManager.getCurrentDriver());
+				break;
+			case "Square":
+				complexCommandFactory = new ComplexCommandFactory();
+				complexCommandFactory.getSquareCommand().execute(driverManager.getCurrentDriver());
+				break;
+			case "Rectangle":
+				complexCommandFactory = new ComplexCommandFactory();
+				complexCommandFactory.getRectangleCommand().execute(driverManager.getCurrentDriver());
+				break;
+			case "Triangle":
+				complexCommandFactory = new ComplexCommandFactory();
+				complexCommandFactory.getTriangleCommand().execute(driverManager.getCurrentDriver());
+				break;
+		}
```

Comment:
Menu action should not depend on its caption. Enum or separate listeners would be better. At least you should consider constants.